src/Lecture1.hs

```diff
@@ -94,7 +101,16 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> String -> String
+subString start end str = result start end str
```

Comment:
I think that the helper `result` function is redundant here 🤔 

You pass all the same arguments to `result` without changing them as `subString` takes. So you can get rid of `result` and put its implementation on the top level here.